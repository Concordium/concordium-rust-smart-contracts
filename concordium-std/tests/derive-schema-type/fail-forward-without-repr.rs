//! Ensure `derive(SchemaType)` fails when using 'forward' attribute without a
//! 'repr' attribute.
use concordium_std::*;

#[derive(SchemaType)]
enum Count {
    One {
        field: u32,
    },
    #[concordium(forward = [5, 6])]
    Two(Inner),
}

#[derive(SchemaType)]
#[concordium(repr(u8))]
enum Inner {
    #[concordium(tag = 5)]
    Alpha {
        balance: u32,
    },
    #[concordium(tag = 6)]
    Beta(u16),
}

fn main() {}
