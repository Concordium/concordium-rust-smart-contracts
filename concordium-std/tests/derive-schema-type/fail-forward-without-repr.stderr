error: 'repr(..)' attribute must be set on the type when using 'forward' attribute
 --> tests/derive-schema-type/fail-forward-without-repr.rs:5:10
  |
5 | #[derive(SchemaType)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `SchemaType` (in Nightly builds, run with -Z macro-backtrace for more info)
